predicates/source/source.go

```diff
@@ -86,11 +99,51 @@ func (s *spec) Create(args []interface{}) (routing.Predicate, error) {
 }
 
 func (p *predicate) Match(r *http.Request) bool {
-	src := snet.RemoteHost(r)
+	var src net.IP
+	if p.fromLast {
+		src = snet.RemoteHostFromLast(r)
+	} else {
+		src = snet.RemoteHost(r)
+	}
+
 	for _, acceptedNet := range p.acceptedSourceNets {
 		if acceptedNet.Contains(src) {
 			return true
 		}
 	}
 	return false
 }
+
+func NewFromLast() routing.PredicateSpec { return &specLast{} }
+
+func (s *specLast) Name() string {
+	return Name
+}
+
+func (s *specLast) Create(args []interface{}) (routing.Predicate, error) {
+	if len(args) == 0 {
```

Comment:
this duplicate code could be avoided, if the spec would be the same for the two flavors with the bool field passed to the predicate instance. The separate New() and NewFromLast() functions should stay but return the same spec with this bool field difference.